### PR TITLE
Add api documentation and testing against api documentation

### PR DESCRIPTION
**Pull request by jasalisbury**
**Description of change**

Add [Open api 3](https://swagger.io/specification/) api docs in `docs/openapi`. Api documentation uses ~(speccy)[https://github.com/wework/speccy] to lint and combine yaml files (so we don't have a single huge yaml file)~ [Redoc](https://github.com/Redocly/redoc) to serve API documentation locally

**How to test**

You'll need to update dependencies (so `yarn deps` or `yarn docker:deps`). Run `yarn docs:serve` (or `yarn docker:start`) and browse to `localhost:5000` to view the API documentation. ~You can run the docs linter with `yarn docs:lint`.~

**Checklist**

~- [n/a] Meets issue criteria~
- [x] Code tested
~- [n/a] Meets accessibility standards (WCAG 2.1 Levels A, AA)~
- [x] Documentation updated
    - API methods
    - Boundary and Data Flow Diagrams
    - [Architectural Decision Records](https://adr.github.io/) written for major infrastructure decisions with the [Nygard template](https://github.com/joelparkerhenderson/architecture_decision_record/blob/master/adr_template_by_michael_nygard.md)
    - OSCAL templates completed when security controls are implemented or modified

### CODE REVIEW

**Comment by kryswisnaskas** [comment]
Nice! 🎉 

Do you think we could also add an example for `POST` ?

So far I was able to run the doc server and see the documentation 👍 
 Having issues with running the `yarn api:test` locally (haven't tried the docker version yet).

**Comment by jasalisbury** [comment]
> Nice! 
> 
> Do you think we could also add an example for `POST` ?
> 
> So far I was able to run the doc server and see the documentation 
> Having issues with running the `yarn api:test` locally (haven't tried the docker version yet).

I added an example for `POST`. Also I added `dredd` to dev dependencies (before I installed it globally) so you should be able to run tests locally after running `yarn deps`

**Comment by kryswisnaskas** [comment]
Thanks. The doc page now shows both `GET `and `POST`. Looks good!

Still getting a couple of errors on `api:test`

Here is more detail on what I am seeing:

`yarn docker:start `- Works ✔️ 
`yarn docker:api:test` - 'concurrently' is not recognized as an internal or external command,
`yarn start `- Works ✔️ 
`yarn docs:serve` - Works ✔️ 
`yarn docs:lint` - Works ✔️ 
`yarn api:test `- error: The "path" argument must be of type string. Received undefined TypeError [ERR_INVALID_ARG_TYPE]: The "path

Also, assuming there will be an ADR written?

File: package.json [hunk 1]
```diff
@@ -19,11 +19,16 @@
     "lint:ci": "eslint -f eslint-formatter-multiple src",
     "lint:all": "yarn lint:ci && yarn --cwd frontend lint:ci",
     "clean": "rm -rf coverage reports frontend/coverage frontend/reports frontend/build",
+    "api:test": "yarn docs:build && npx dredd --server 'yarn server'",
```
Comment (kryswisnaskas) [inline package.json:22]:
```suggestion
    "api:test": "yarn docs:build && npx dredd 'yarn server'",
```

Comment (kryswisnaskas) [reply package.json:22]: Without `--server` `api:test` works for me locally.

Comment (jasalisbury) [reply package.json:22]: I'm having a hard time getting this one to play nicely with running on a windows and linux host and at the same time being able to be ran from docker. Maybe Dredd is more trouble then it is worth at this point? I was on the edge on if I thought documentation testing was worth it and I'm leaning more towards thinking it isn't now.

Comment (kryswisnaskas) [reply package.json:22]: Assuming `npx dredd 'yarn server'` doesn't work for you right?

It is nice that it tests the documentation, but we could remove it if you have second thoughts about it and possibly include it again later if needed.

Comment (jasalisbury) [reply package.json:22]: One last thing to try... can you tell me if `npx dredd --config ./dredd.yml` works for you?

**Comment by jasalisbury** [comment]
I've removed dredd and added an API documentation ADR. Also touched up the formatting on some of the older ADRs to be more consistent.

File: docs/openapi/index.yaml [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+tags:
+  - name: status
+info:
+  version: 0.0.1
+  title: TTADP
+  contact:
+    email: josh@adhocteam.us
+servers:
+  - url: http://localhost:8080
+  - url: http://backend:8080
```
Comment (kryswisnaskas) [inline docs/openapi/index.yaml:11]: Should this be configurable for different environments?

**Review by kryswisnaskas — COMMENTED** [review]
Great! I also like how the code is re-organized in this PR.

File: docs/openapi/index.yaml (hunk 2 above)
Comment (jasalisbury) [reply docs/openapi/index.yaml:11]: Good catch, I'll add the dev environment to this list. We will want to add staging and prod once we know their URLs

**Review by kryswisnaskas — APPROVED** [review]

**Comment by jasalisbury** [comment]
Had to take out speccy, it is no longer maintained. Should have guessed because it was made by wework... Anyways we now use a library called `ReDoc` to serve the API documentation.

File: docs/openapi/index.yaml [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+  - url: https://tta-smarthub-sandbox.app.cloud.gov
```
Comment (kryswisnaskas) [inline docs/openapi/index.yaml:12]: Should we also add the other environment servers here (dev, staging, prod) ? Or perhaps incorporate variables in urls?

**Review by kryswisnaskas — APPROVED** [review]
Looking at this again after the change. Overall looks good. Left one comment for the `server ` option. 

Other - I needed to install redoc-cli locally to be able to run the server. The docker version worked well though.

**Pull request merged**